recipe_scrapers/_abstract.py

```diff
@@ -1,7 +1,8 @@
-# mypy: disallow_untyped_defs=False
+from __future__ import annotations
```

Comment:
What's the reason for adding this `import` @sbdchd?

Reply:
`from __future__ import annotations` makes annotations lazy so we can use new syntax and it also means we don't have any runtime overhead for them

Reply:
Hrm, OK - is that the same thing as [PEP-563](https://peps.python.org/pep-0563/)?

(if so: it looks like that was [reverted from Python 3.10](https://mail.python.org/archives/list/python-dev@python.org/thread/CLVXXPQ2T2LQ5MP2Y53VVQFCXYWQJHKZ/) -- and either way, I think it could be risky to begin using a feature like that here until it's well-proven.  it's probably useful, but I'd prefer to be cautious)